module/VuFind/tests/unit-tests/src/VuFindTest/Navigation/AdminMenuTest.php

```diff
@@ -64,49 +66,44 @@ public function testDefaultMenuAllCheckMethodsReturnFalse()
     {
         $menu = $this->getAdminMenu(
             AdminMenu::getDefaultMenuConfig(),
-            $this->getCheckMethods(false)
+            $this->getAdminMenuCheckMethods(false)
         )->getMenu();
         $this->assertCount(7, $menu['Admin']['MenuItems']);
     }
 
     /**
-     * Get mock AdminMenu.
+     * Data provider for testRequiredConfiguration
      *
-     * @param array $config       Configuration to use
-     * @param array $checkMethods Values to return for specific check methods
-     *
-     * @return AdminMenu
+     * @return \Iterator<string, array>
      */
-    protected function getAdminMenu(
-        array $config = [],
-        array $checkMethods = [],
-    ): AdminMenu {
-        $adminMenu = $this->getMockBuilder(AdminMenu::class)
-            ->setConstructorArgs(
-                [
-                    $config,
-                    false,
-                ]
-            )
-            ->onlyMethods(array_keys($this->getCheckMethods()))
-            ->getMock();
-        foreach ($this->getCheckMethods() as $checkMethod => $default) {
-            $adminMenu->method($checkMethod)->willReturn($checkMethods[$checkMethod] ?? $default);
-        }
-        return $adminMenu;
+    public static function requiredConfigurationProvider(): \Iterator
+    {
+        yield 'Missing menu item settings' => [
+            [
+                'Admin' => [
+                    'MenuItems' => [[]],
+                ],
+            ],
+        ];
     }
 
     /**
-     * Get all check methods.
+     * Test required configuration.
      *
-     * @param bool $value Value for the check methods to return
+     * @param array  $config                      Account menu configuration
+     * @param string $expectedExceptionClass      Expected exception class
+     * @param string $expectedExceptionMsgMatches Expected exception message regexp
      *
-     * @return array
+     * @return void
      */
-    protected function getCheckMethods(bool $value = true): array
-    {
-        return [
-            'checkShowOverdrive' => $value,
-        ];
+    #[\PHPUnit\Framework\Attributes\DataProvider('requiredConfigurationProvider')]
+    public function testRequiredConfiguration(
+        array $config,
+        string $expectedExceptionClass = BadConfig::class,
+        string $expectedExceptionMsgMatches = '/^Missing required setting: /'
+    ) {
```

Comment:
Might as well include type:

```suggestion
    ): void {
```